### PR TITLE
Add documentation for tasks

### PR DESCRIPTION
**Pull request by planger**
Fixes https://github.com/eclipse-theia/theia-website/issues/199

Contributed on behalf of STMicroelectronics

Signed-off-by: Philip Langer <planger@eclipsesource.com>

### CODE REVIEW

File: src/docs/tasks.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
+---
+title: Tasks
+---
+
+# Tasks
+
+Eclipse Theia users can execute tasks in the context of a workspace to automate certain steps in their workflow.
```
Comment (vince-fugnitto) [inline src/docs/tasks.md:7]: There is also support for `user`-level tasks (https://github.com/eclipse-theia/theia/issues/7446) but we don't necessarily need to mention them at this time.

File: src/docs/tasks.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+## Example: Task Providers and Task Resolvers
+
+In the following example, we will contribute a custom task provider, which provides a custom task.
+Further, we will add a custom task resolver that will enhance the task configuration before execution.
+Finally, we will contribute a custom task runner that executes our provided task.
+
+Task providers and task resolvers are contributed via an implementation of `TaskContribution`.
+Like all contributions, it must be bound in the respective front end module as shown below.
+If you are not yet familiar with contribution points in Theia or the use of dependency injection, please consider this guide on Services and Contributions.
```
Comment (vince-fugnitto) [inline src/docs/tasks.md:40]: Did you mean to add a link for `services and contributions`?

```suggestion
If you are not yet familiar with contribution points in Theia or the use of dependency injection, please consider this guide on [Services and Contributions](https://theia-ide.org/docs/services_and_contributions/).
```

File: src/docs/tasks.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+   registerRunner(runners: TaskRunnerRegistry): void {
+       runners.registerRunner('myTaskType', this.myTaskRunner);
+   }
+}
+```
+
+Task runners need to implement the interface `TaskRunner`.
+The function `run`receives the `TaskConfiguration` when a task is triggered and is responsible for actually running the operation.
+In our example, we instantiate a custom implementation of `Task` called `MyTask` and execute it with the current configuration.
+Using the existing interface `Task` allows us to connect the task to the `TaskManager` which will show visual feedback in the workbench during taks execution (see screenshot below).
```
Comment (vince-fugnitto) [inline src/docs/tasks.md:118]: minor typo:

```suggestion
Using the existing interface `Task` allows us to connect the task to the `TaskManager` which will show visual feedback in the workbench during tasks execution (see screenshot below).
```

File: src/docs/tasks.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+As can be seen in console output, the task starts and finishes 5000 ms later and it prints the custom variable that has been added by the custom resolver.
+
+```
+root INFO Start running custom task: 42
+root INFO Finished running custom task: 42
+```
+
+## Task Definitions
+
+Extensions can also omit  contributing a task provider, but only contribute a runner (and optionally a resolver).
```
Comment (vince-fugnitto) [inline src/docs/tasks.md:167]: minor: extra whitespace

```suggestion
Extensions can also omit contributing a task provider, but only contribute a runner (and optionally a resolver).
```

File: src/docs/tasks.md (hunk 1 above)
Comment (planger) [reply src/docs/tasks.md]: Thanks, good input! I removed "in the context of a workspace" as it is vague anyway and added a sentence at the end that tasks can be specified in the workspace or on user level.

**Comment by planger** [comment]
@vince-fugnitto thanks a lot for your valuable feedback! I committed your suggestions and rephrased the first paragraphs to take user-level tasks into account.

File: src/docs/tasks.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+   protected readonly myTaskRunner: MyTaskRunner;
+
+   registerRunner(runners: TaskRunnerRegistry): void {
+       runners.registerRunner('myTaskType', this.myTaskRunner);
+   }
+}
+```
+
+Task runners need to implement the interface `TaskRunner`.
+The function `run`receives the `TaskConfiguration` when a task is triggered and is responsible for actually running the operation.
```
Comment (vince-fugnitto) [inline src/docs/tasks.md:116]: minor: missing whitespace

```suggestion
The function `run` receives the `TaskConfiguration` when a task is triggered and is responsible for actually running the operation.
```

File: src/docs/tasks.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+   }
+   …
+}
+```
+
+As you can see in the screenshot below, our custom task is running for 5000 ms (as we set a timeout in `MyTask`) and then it stops.
+
+<img src="/running-custom-task.gif" alt="Running custom task" style="max-width: 702px">
+
+As can be seen in console output, the task starts and finishes 5000 ms later and it prints the custom variable that has been added by the custom resolver.
```
Comment (vince-fugnitto) [inline src/docs/tasks.md:158]: minor: I believe `the` might be missing?

```suggestion
As can be seen in the console output, the task starts and finishes 5000 ms later and it prints the custom variable that has been added by the custom resolver.
```

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me, I just have two minor comments :+1:

**Pull request merged**